### DISCUSSION TITLE
Multi-user?

### DISCUSSION THREAD

**Discussion by progettoautomazione** [gitpod-io/openvscode-server#119] [Q&A] [answered]
Hi everyone, we are currently using AWS Cloud9 editor, as web-based IDE that connects to our codebase server by SSH and allows me and my colleague to edit code in real time, saving changes directly on the filesystem where the codebase is hosted.
Is this something that can be accomplished by OpenVSCode Server? If yes I'm looking forward to step away from AWS Cloud9 because it's getting seriously bugged, plus it's not extendible like VS. Thanks

**Comment by akosyakov** [comment, marked as answer]
> Is this something that can be accomplished by OpenVSCode Server? If yes I'm looking forward to step away from AWS Cloud9 because it's getting seriously bugged, plus it's not extendible like VS. Thanks

VS Code is not really designed to be multi tenant, I guess you can get something like that if you disable all possible auths and so on. But I would recommend to look for full multi tenant solutions like Gitpod or Codespaces. For Gitpod there is self-hosted variant which is free up to 10 users: https://www.gitpod.io/self-hosted

**Reply by progettoautomazione** [reply]
Thanks @akosyakov I would like to go with Gitpod or Codespaces but as far as I know they are based on Kubernetes and this would be a hyper-structure to me. I "just" need a tool that access locally or by ssh to a folder on my filesystem, so that it can edit/create/delete files. Multi-users. Exactky like cloud9, that other than allowing users to create containers (doker), it can simply connect to a folder and let the editor work on this folder. Easier and lighter than setting up a kubernetes/docker thing...

**Reply by akosyakov** [reply]
Have you tried to run it already? Generally anyone with a link can access the server, user settings are stored in local browser per origin, so not shared with other users. User extensions can be the issue. You may try to start a server several times on different ports for different linux users to avoid this issue.
